### PR TITLE
core: enable finalizers on extmod classes

### PR DESCRIPTION
**Pull request by matejcik**
C-defined classes that have a `__del__` attribute can allocate the objects with a finalizer flag, and the GC will call `__del__` when deallocating.
Unfortunately, this must be done explicitly.

This patch makes sure that all classes with `__del__` are allocated with the finalizer flag set.

### CODE REVIEW

**Comment by matejcik** [comment]
related: #705, #981

File: core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h [hunk 1]
```diff
@@ -64,8 +65,11 @@ STATIC mp_obj_t mod_trezorcrypto_secp256k1_context_make_new(
   const size_t secp256k1_ctx_size = secp256k1_context_preallocated_size(
       SECP256K1_CONTEXT_SIGN | SECP256K1_CONTEXT_VERIFY);
 
-  mp_obj_secp256k1_context_t *o = m_new_obj_var_maybe(
-      mp_obj_secp256k1_context_t, uint8_t, secp256k1_ctx_size);
+  // there is no finaliser-enabled "m_new_obj_var_maybe" so we have to call
```
Comment (prusnak) [inline core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h:68]: Maybe wrapping this into a macro called `m_new_obj_var_maybe_with_finalizer` or something similar would be less cryptic?

File: core/embed/extmod/modtrezorcrypto/modtrezorcrypto-rfc6979.h [hunk 2]
```diff
@@ -40,7 +40,7 @@ STATIC mp_obj_t mod_trezorcrypto_Rfc6979_make_new(const mp_obj_type_t *type,
                                                   size_t n_args, size_t n_kw,
                                                   const mp_obj_t *args) {
   mp_arg_check_num(n_args, n_kw, 2, 2, false);
-  mp_obj_Rfc6979_t *o = m_new_obj(mp_obj_Rfc6979_t);
+  mp_obj_Rfc6979_t *o = m_new_obj_with_finaliser(mp_obj_Rfc6979_t);
```
Comment (andrewkozlik) [inline core/embed/extmod/modtrezorcrypto/modtrezorcrypto-rfc6979.h:43]: `rfc6979` doesn't have a `__del__()` method. I guess it should have one to memzero the context. However, I can't actually see any use of rfc6979 in our code.

Comment (prusnak) [reply core/embed/extmod/modtrezorcrypto/modtrezorcrypto-rfc6979.h:43]: Let's drop the thing if it's not being used.

Comment (matejcik) [reply core/embed/extmod/modtrezorcrypto/modtrezorcrypto-rfc6979.h]: done in 6cb8eed

File: core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h (hunk 1 above)
Comment (matejcik) [reply core/embed/extmod/modtrezorcrypto/modtrezorcrypto-secp256k1_zkp.h]: done in  6fd14cd

**Pull request merged**